src/Resources/Resources.Test/ScenarioTests/DeploymentScriptsTests.ps1

```diff
@@ -298,6 +342,17 @@ function Test-GetDeploymentScriptLog-Cli
 		Assert-AreEqual $deploymentScriptName $getLogByResourceId.DeploymentScriptName
 
 
+		# Test - GetLogByDeploymentScriptResourceId - WithTailParameter
+		$getLogByResourceIdWithTail = Get-AzDeploymentScriptLog -DeploymentScriptResourceId $resourceId -Tail $TailInteger
+		$Tail = [math]::min( ($getLogByResourceId.Log -split '\n').count, $TailInteger)
+
+		#Assert
+		Assert-NotNull $getLogByResourceIdWithTail
+		Assert-NotNull $deploymentScriptName $getLogByResourceIdWithTail.Log
+		Assert-AreEqual $deploymentScriptName $getLogByResourceIdWithTail.DeploymentScriptName
+		Assert-AreEqual ($getLogByResourceIdWithTail.Log -split '\n').count $Tail
```

Comment:
>Assert-AreEqual ($getLogByResourceIdWithTail.Log -split '\n').count $Tail [](start = 2, length = 73)

This is not really testing Tail for the CLI logs (we discussed this offline).  You can use '\r' to split the log.